src/wp-includes/template.php

```diff
@@ -698,20 +714,24 @@ function get_attachment_template() {
  * @return string The template filename if one is located.
  */
 function locate_template( $template_names, $load = false, $load_once = true, $args = array() ) {
-	$stylesheet_path = get_stylesheet_directory();
-	$template_path   = get_template_directory();
-	$is_child_theme  = $stylesheet_path !== $template_path;
+	global $wp_stylesheet_path, $wp_template_path;
+
+	if ( ! isset( $wp_stylesheet_path ) || ! isset( $wp_template_path ) ) {
+		wp_set_template_globals();
+	}
+
+	$is_child_theme = $wp_stylesheet_path !== $wp_template_path;
```

Comment:
Smart 😉 Instead of `is_child_theme()` use local variables.

Reply:
Actually, would probably be better to apply these changes to `is_child_theme()` since that function was changed in 6.4 when we deprecated the constants.